cpp/src/Ice/RouterInfo.cpp

```diff
@@ -229,29 +229,7 @@ IceInternal::RouterInfo::setClientEndpoints(const optional<ObjectPrx>& proxy, bo
     if (_clientEndpoints.empty())
     {
         _hasRoutingTable = hasRoutingTable;
-        if (!proxy)
-        {
-            //
-            // If getClientProxy() return nil, use router endpoints.
-            //
-            _clientEndpoints = _router->_getReference()->getEndpoints();
-        }
-        else
-        {
-            ObjectPrx clientProxy = proxy->ice_router(nullopt); // The client proxy cannot be routed.
```

Comment:
This call was totally useless as far as I can tell. clientProxy is a local variable.